edx_event_bus_kafka/internal/consumer.py

```diff
@@ -132,6 +133,24 @@ def _shut_down(self):
         """
         self._shut_down_loop = True
 
+    def _db_needs_reconnect(self):
+        """
+        Determine if we need to try to reconnect to the database
+
+        Returns:
+            True if we have a db connection but it is unusable
+            False otherwise (if we don't have a connection or our current connection is fine)
+        """
+        return connection.connection and not connection.is_usable()
+
+    def _reconnect_to_db(self):
+        """
+        Pass-through to connection.connect()
+
+        Moved out for easier testing (mocking connections in pytest is hard)
```

Comment:
[Nit-ish] Mocking is already once removed from reality, and mocking the wrappers seems even further removed. If you simply just patch/mock the top-level `connection`, can you use code like the following?:
```
mock_connection.connection = None
mock_connection.connect.return_value = True
mock_connection.is_usable.return_value = True
etc.
```

Reply:
Mocking the top level connection turns out to wreak havoc on the tests, which is why I separated these wrappers out. I can try again but I don't think it's worth the development hours.

Reply:
[wontfix]

Reply:
I hope you don't mind, but I made a PR for this: https://github.com/openedx/event-bus-kafka/pull/117